### ISSUE TITLE
[INSTALL] Install the libraries for ufs-weather-model develop branch on wcoss2

### ISSUE THREAD

**Issue by junwang-noaa** [NOAA-EMC/hpc-stack#432]
Note: Please search to see if an issue already exists for the software you are requesting to be installed.

**Which software in the stack would you like installed?**
Please install the libraries on wcoss2 to support ufs-weather-model develop branch.

**What is the version/tag of the software?**
 jasper/2.0.22 
png/1.6.35 
pio/2.5.2 
esmf/8.3.0b09
esmf/8.3.0b09-debug 
fms/2022.01
g2/3.4.2 
gftl-shared/v1.3.3 
yafyaml/v0.5.1 
mapl/2.11.0-esmf-8.3.0b09

**What compilation options would you like set?**
Give a detailed list of compilation options, compiler and MPI flavors and versions, etc.

**Which machines would you like to have the software installed?**
wcoss2

**Any other relevant information that we should know to correctly install the software??**
Please describe in as much detail as possible.

**Additional context**
Add any other context or screenshots about the install request here.

**Comment by kgerheiser** [comment]
FMS needs a newer version. `2021.04` is the latest.

ESMF is available. Installed in `/para` right now (along with FMS 2021).

```
export HPC_OPT=/apps/ops/para/libs
module use /apps/ops/para/libs/modulefiles/compiler/intel/19.1.3.304/
module use /apps/ops/para/libs/modulefiles/mpi/intel/19.1.3.304/cray-mpich/8.1.7/
module load intel craype cray-mpich/8.1.7

module load esmf/8.3.0b09
```

PIO and MAPL (along with associated dependencies) are being reviewed.


And for some of these could we update the develop branch to use `jasper/2.0.25`, `g2/3.4.5`, and `libpng/1.6.37`?

Jasper and g2 are already available everywhere. And we can add `libpng/1.6.37` to hpc-stack installations.

**Comment by junwang-noaa** [comment]
@kgerheiser Thank you very much for the information, please let us know when libpng/1.6.37 is available, we can use fms/2021.04 for testing now, but since we are moving to fms 2022.01 this week, so we will need it when updating ufs-weather-model with wcoss2 porting change.  

@DusanJovic-NOAA FYI.

**Comment by jkbk2004** [comment]
@junwang-noaa @kgerheiser I agree to merge in fms2022.01 PR first. Then we can address other updates: libpng, hdf5, netcdf.

**Comment by kgerheiser** [comment]
@junwang-noaa I'll see about getting 2022.01 installed on WCOSS2.

@jkbk2004 yes, that's fine. There are no HDF5/NetCDF updates though.

I think UFS should be updated to use newer g2 (3.4.5), Jasper (2.0.25), PIO (2.5.3), libpng (1.6.37) in the near future, and then they'll be the more up-to-date and consistent with what WCOSS2 already has, and our stack builds by default.

I would expect no differences when using these newer versions.

**Comment by WenMeng-NOAA** [comment]
@arunchawla-NOAA @kgerheiser I would like to get some clarifications. The NCEPlibs installed under /apps/ops/para/libs/modulefiles/mpi/intel/19.1.3.304/cray-mpich/8.1.7 and /apps/ops/para/libs/modulefiles/compiler/intel/19.1.3.304 would support porting ufs-weather-model and UPP develop branch on WCOSS2? These libs are installed at dev site maintained by EIB?

**Comment by DusanJovic-NOAA** [comment]
> @junwang-noaa I'll see about getting 2022.01 installed on WCOSS2.
> 
> @jkbk2004 yes, that's fine. There are no HDF5/NetCDF updates though.
> 
> I think UFS should be updated to use newer g2 (3.4.5), Jasper (2.0.25), PIO (2.5.3), libpng (1.6.37) in the near future, and then they'll be the more up-to-date and consistent with what WCOSS2 already has, and our stack builds by default.
> 
> I would expect no differences when using these newer versions.

Ok. Please make sure g2 (3.4.5), Jasper (2.0.25), PIO (2.5.3), libpng (1.6.37)  are installed everywhere and we'll update UFS.

**Comment by DusanJovic-NOAA** [comment]
`/apps/ops/para/libs/modulefiles/stack` is empty.

**Comment by junwang-noaa** [comment]
@KyleNevins-NOAA May I ask if libpng/1.6.37 is available on wcoss2 now? Also, please see the message from Dusan: "/apps/ops/para/libs/modulefiles/stack is empty."

**Comment by DusanJovic-NOAA** [comment]
@kgerheiser Is there a plan to install hpc-stack in /apps/ops/para/libs such that we can:

```
module use /apps/ops/para/libs/modulefiles/stack

module load hpc/1.2.0
module load hpc-intel/19.1.3.304
module load hpc-cray-mpich/8.1.7
```
In the above comment you suggested that we update pio to 2.5.3, but that version is not available. Any updates on gftl-shared, yafyaml and mapl? Also 8.3.0b09-debug is missing. 

General question, will this location (/apps/ops/para/libs/) hold all hpc-stack libraries required by the ufs-weather-model, as on any other supported platform? Or just a subset?

**Comment by WenMeng-NOAA** [comment]
I echo @DusanJovic-NOAA 's suggestion. From our UPP perspective, it would be better to use libs from hpc-stack for UPP develop branch.

**Comment by Hang-Lei-NOAA** [comment]
That is not the NCO's way. They only use hpc-stack for the installation,
and then copy, move into their structure.

**Comment by DusanJovic-NOAA** [comment]
> That is not the NCO's way. They only use hpc-stack for the installation, and then copy, move into their structure.

@Hang-Lei-NOAA Are you saying that libraries in /apps/ops/para/libs are not installed by the hpc-stack team (nceplibs team)?

**Comment by arunchawla-NOAA** [comment]
NCO uses hpc-stack but installs by themselves. Right now pio is going through NCO review, as soon as that is done it will be installed. After that the NASA libraries will be installed. Once pio is there will you be able to test the ufs-weather-model build ? Not withstanding the gocart related libraries

**Comment by DusanJovic-NOAA** [comment]
Is somebody from NCO looking at this issue?

**Comment by edwardhartnett** [comment]
The ufs-weather-model build uses PIO?

**Comment by Hang-Lei-NOAA** [comment]
We (NCEPLIBS team in EMC) do not have the right to install. We only pass it
to NCO.
Many places in NCO do the review. GDIT review the third party libs. NCO
Steven's team review our nceplibs.

**Comment by arunchawla-NOAA** [comment]
@DusanJovic-NOAA yes, @edwardhartnett yes pio is used by cmeps and esmf also now
